linode/manager
# upcoming: [M3-8287] - Add New Search Parser

## Description 📝

- Kicks off _Search v2 ✨_ by adding a "search query to API filter" parser
  - New search parser is in it's own package to prepare for micro-frontends 📦 
  - Keep in mind, this new code isn't "final" at all. We will iterate and document this over time as we improve search throughout Cloud Manager 🔍 
- Implements new parser in Linode Create v2 for StackScript filtering 📃 
  - This achieves parity with the old flow plus all of the new operations like and, or, etc..

## Preview 📷

![Screenshot 2024-06-25 at 5 39 39 PM](https://github.com/linode/manager/assets/115251059/9a05e998-51ca-45d6-8106-c0660bc1b4e2)

## How to test 🧪

- Check basics of new package structure 📦 
- Check functionality of new search 🔍 
  - Turn on Linode Create v2
  - Test search on `http://localhost:3000/linodes/create?type=StackScripts`
  - Verify search functionality is better than or equivalent to Linode Create v1

## As an Author I have considered 🤔
